File: leetcode-daily/17-January/Find the Largest Area of Square Inside Two Rectangles.java
class Solution {
    public long largestSquareArea(int[][] bottomLeft, int[][] topRight) {
        

        long side = 0L;
        for(int i=0;i<bottomLeft.length;i++){
            for(int j=i+1;j<bottomLeft.length;j++){
                long width = Math.min(topRight[i][0],topRight[j][0])-Math.max(bottomLeft[i][0],bottomLeft[j][0]);
                long height = Math.min(topRight[i][1],topRight[j][1])-Math.max(bottomLeft[i][1],bottomLeft[j][1]);
                side = Math.max(side,Math.min(width,height));
            }
        }
        return side*side*1L;
    }
}

File: leetcode-daily/17-January/Find the Largest Area of Square Inside Two Rectangles.cpp
class Solution {
public:
    long long largestSquareArea(vector<vector<int>>& b, vector<vector<int>>& t){

        int n = b.size();
        long long ans = 0;
        for(int i=0;i<n;i++){
            for(int j=i+1;j<n;j++){
                long long width = min(t[i][0],t[j][0])-max(b[i][0],b[j][0]);
                long long height = min(t[i][1],t[j][1])-max(b[i][1],b[j][1]);
                ans = max(ans,min(width,height));
            }
        }
        return ans*ans*1LL;
    }
};